server/src/main/java/org/opensearch/common/settings/AbstractScopedSettings.java

```diff
@@ -121,8 +122,13 @@ protected AbstractScopedSettings(
                 keySettings.putIfAbsent(setting.getKey(), setting);
             }
         }
-        this.complexMatchers = Collections.unmodifiableMap(complexMatchers);
-        this.keySettings = Collections.unmodifiableMap(keySettings);
+        if (FeatureFlags.isEnabled(FeatureFlags.EXTENSIONS)) {
```

Comment:
I think this feature should not be related to `EXTENSIONS`, right?

Reply:
+1, we can just enable this IMO